zerver/lib/bugdown/__init__.py

```diff
@@ -1867,16 +1867,16 @@ def build_block_parser(self) -> markdown.util.Registry:
         # quote - replaced by ours
         parser = markdown.blockprocessors.BlockParser(self)
         parser.blockprocessors.register(markdown.blockprocessors.EmptyBlockProcessor(parser), 'empty', 85)
-        if not self.getConfig('code_block_processor_disabled'):
-            parser.blockprocessors.register(markdown.blockprocessors.CodeBlockProcessor(parser), 'code', 80)
-        parser.blockprocessors.register(HashHeaderProcessor(parser), 'hashheader', 78)
+        parser.blockprocessors.register(HashHeaderProcessor(parser), 'hashheader', 80)
         # We get priority 75 from 'table' extension
         parser.blockprocessors.register(markdown.blockprocessors.HRProcessor(parser), 'hr', 70)
         parser.blockprocessors.register(OListProcessor(parser), 'olist', 68)
         parser.blockprocessors.register(UListProcessor(parser), 'ulist', 65)
         parser.blockprocessors.register(ListIndentProcessor(parser), 'indent', 60)
-        parser.blockprocessors.register(BlockQuoteProcessor(parser), 'quote', 55)
-        parser.blockprocessors.register(markdown.blockprocessors.ParagraphProcessor(parser), 'paragraph', 50)
+        if not self.getConfig('code_block_processor_disabled'):
+            parser.blockprocessors.register(markdown.blockprocessors.CodeBlockProcessor(parser), 'code', 55)
+        parser.blockprocessors.register(BlockQuoteProcessor(parser), 'quote', 50)
+        parser.blockprocessors.register(markdown.blockprocessors.ParagraphProcessor(parser), 'paragraph', 45)
         return parser
```

Comment:
What is the ordering between the code block, quote, and paragraph processors in `python-markdown` itself?  Is this taking us closer or further from their ordering?

Maybe it's worth adding a block comment at the top showing the ordering used by upstream so that we can start documenting places where we intentionally diverge from that behavior and adjusting places where we don't?

Reply:
Good catch! This is the (relatively) correct change, but the eventual result is:

```
--- /home/rohitt/ramdisk/upstream
+++ /home/rohitt/ramdisk/zulip
@@ -1,10 +1,9 @@
 EmptyBlockProcessor
-ListIndentProcessor
-CodeBlockProcessor
 HashHeaderProcessor
-SetextHeaderProcessor
 HRProcessor
 OListProcessor
 UListProcessor
+ListIndentProcessor
+CodeBlockProcessor
 BlockQuoteProcessor
 ParagraphProcessor
```

I'm amending this commit to instead move ListIndentProcessor above and see if anything fails.